src/smart-components/workspaces/WorkspaceListTable.tsx

```diff
@@ -107,6 +118,47 @@ const WorkspaceListTable = () => {
     setIsDeleteModalOpen(!isDeleteModalOpen);
   };
 
+  const isValidEditType = (workspace: Workspace) => {
+    switch (workspace.type) {
+      case 'root':
+        return false;
+      case 'default':
+        return true;
+      case 'ungrouped-hosts':
+        return false;
+      case 'standard':
+        return true;
+    }
+  };
+
+  const isValidDeleteType = (workspace: Workspace) => {
+    switch (workspace.type) {
+      case 'root':
+        return false;
+      case 'default':
+        return false;
+      case 'ungrouped-hosts':
+        return false;
+      case 'standard':
+        return true;
+    }
+  };
+
+  const canModify = (workspace: Workspace, action: 'edit' | 'delete') => {
+    if (
+      userPermissions.permission === 'inventory:groups:write' &&
+      (userPermissions.resourceDefinitions.length === 0 || userPermissions.resourceDefinitions.includes(workspace.id))
+    ) {
+      if (action === 'edit' && isValidEditType(workspace)) {
+        return true;
+      } else if (action === 'delete' && isValidDeleteType(workspace)) {
+        return true;
+      }
+    } else {
+      return false;
+    }
```

Comment:
Let's add default return statement here
```suggestion
    }
    return false;
```